tools/rpkg/tests/testthat/test_backend-dbplyr__duckdb_connection.R

```diff
@@ -0,0 +1,185 @@
+# helper function to skip tests if we don't have the suggested package
+# https://github.com/DyfanJones/noctua/blob/master/tests/testthat/helper.R
+skip_if_package_not_available <- function(pkg) {
+  if (!nzchar(system.file(package = pkg)))
+    skip(sprintf("`%s` not available for testing", pkg))
+}
+
+skip_if_no_R4 <- function() {
+  if (R.Version()$major<4) {
+    skip("R 4.0.0 or newer not available for testing")
+  }
+}
+
+test_that("dbplyr generic scalars translated correctly", {
+  skip_if_no_R4()
+  skip_if_package_not_available("dbplyr")
+  library(dbplyr)
+  translate <- function(...) translate_sql(...,con=translate_duckdb())
+
+  expect_equal(translate(as.character(1)),               sql(r"{CAST(1.0 AS TEXT)}"))
```

Comment:
So do you suggest that all tests would be changed to snapshot tests? Those certainly could be easier to review, but will that make those tests better? I already used the raw strings introduced in R 4.0 to make the expected results more readable.

Maybe snapshot tests could be added as extra tests here for review purposes if those are really needed?

I have plans on testing equivalence of R and database expressions, but I would like to do that by mimicking the real workflows to show what kind of tasks can be safely conducted at the database. In any case, there is a lot of work with that and I suggest that such tests would be added later on. It would also be nice to get feedback from the other users on their wishes on the "priorities" of required features.

Reply:
Snapshot tests are now added. I agree that these are easier to write and read, but I didn't delete the old tests as they may have some value especially on CRAN.